cli/internal/env/env.go

```diff
@@ -94,90 +125,151 @@ func (evm EnvironmentVariableMap) ToHashable() EnvironmentVariablePairs {
 	})
 }
 
-// GetEnvMap returns a map of env vars and their values from os.Environ
-func GetEnvMap() EnvironmentVariableMap {
-	envMap := make(map[string]string)
-	for _, envVar := range os.Environ() {
-		if i := strings.Index(envVar, "="); i >= 0 {
-			parts := strings.SplitN(envVar, "=", 2)
-			envMap[parts[0]] = strings.Join(parts[1:], "")
+const wildcard = '*'
+const wildcardEscape = '\\'
+const regexWildcardSegment = ".*"
+
+func wildcardToRegexPattern(pattern string) (*string, string) {
+	hasWildcard := false
+	var regexString []string
+	var literalString []string
+
+	var previousIndex int
+	var previousRune rune
+
+	for i, char := range pattern {
+		if char == wildcard {
+			if previousRune == wildcardEscape {
+				// Found a literal *
+
+				// Replace the trailing "\*" with just "*" before adding the segment.
+				literalString = append(literalString, pattern[previousIndex:i-1]+"*")
+				regexString = append(regexString, regexp.QuoteMeta(pattern[previousIndex:i-1]+"*"))
+			} else {
+				// Found a wildcard
+				hasWildcard = true
+
+				// Add in the static segment since the last wildcard. Can be zero length.
+				regexString = append(regexString, regexp.QuoteMeta(pattern[previousIndex:i]))
+
+				// Add a dynamic segment if it isn't adjacent to another dynamic segment.
+				if regexString[len(regexString)-1] != regexWildcardSegment {
+					regexString = append(regexString, regexWildcardSegment)
+				}
+			}
+
+			// Advance the pointer.
+			previousIndex = i + 1
 		}
+		previousRune = char
 	}
-	return envMap
-}
 
-// FromKeys returns a map of env vars and their values from a given set of env var names
-func FromKeys(all EnvironmentVariableMap, keys []string) EnvironmentVariableMap {
-	output := EnvironmentVariableMap{}
-	for _, key := range keys {
-		output[key] = all[key]
+	// Add the last static segment. Can be zero length.
+	literalString = append(literalString, pattern[previousIndex:])
+	regexString = append(regexString, regexp.QuoteMeta(pattern[previousIndex:]))
+
+	// We need the computed literal env value because FOO= is meaningful.
+	var literalValue string
+	if !hasWildcard {
+		literalValue = strings.Join(literalString, "")
+		return &literalValue, strings.Join(regexString, "")
```

Comment:
This is going to go away. This was introduced to prove that the new and old behaviors were identical; though wrong.

The fix to this is to add `env` and `globalEnv` into the correct hash positions.

Reply:
Removed here: https://github.com/nathanhammond/turbo/pull/10